CMakeLists.txt

```diff
@@ -5,44 +5,51 @@
 # accordance with the terms of the Adobe license agreement accompanying
 # it. If you have received this file from a source other than Adobe,
 # then your use, modification, or distribution of it requires the prior
-# written permission of Adobe. 
+# written permission of Adobe.
 
-cmake_minimum_required(VERSION 3.18)
+cmake_minimum_required(VERSION 3.23)
+
+include(FetchContent)
+
+set(CMAKE_OSX_ARCHITECTURES x86_64) # Only build x86 for now
 
 project(hyde)
+
+if (NOT DEFINED CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 20)
+endif ()
+
 set(CMAKE_CXX_EXTENSIONS OFF)
-set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_XCODE_GENERATE_SCHEME OFF)
 
-if(APPLE)
-    # for now we are assuming the user has installed with homebrew.
-    # we detect the install location with brew --prefix llvm
-    execute_process(
-        COMMAND brew --prefix llvm
-        OUTPUT_VARIABLE llvm_to_use
-        OUTPUT_STRIP_TRAILING_WHITESPACE
+if (NOT TARGET llvm::llvm)
```

Comment:
`llvm::llvm` isn't a target provided by the llvm project so I'm not sure it is a meaningful guard. This kind of thing is useful when you want your project to support being built as a subdirectory, but I think it would require a bit more work (especially considering llvm/clang targets are not very well defined). Personally I would postpone that to another commit.

Reply:
The biggest intent here is to keep the dependency from being re-downloaded. If I remove this guard, I'm assuming `FetchContent` is smart enough not to do extra work. Does that sound reasonable?

Reply:
`FetchContent` will only update the pulled repo as needed, not redownload everything all the time, so there is no need for explicitly guarding against that.

Reply:
Removed.